src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -286,7 +286,7 @@ private static Version GetICUVersion()
                               version & 0xFF);
         }
 
-        private static readonly Lazy<bool> _net5CompatFileStream = new Lazy<bool>(() => GetStaticNonPublicBooleanPropertyValue("System.IO.FileStreamHelpers", "UseNet5CompatStrategy"));
+        private static readonly Lazy<bool> _net5CompatFileStream = new Lazy<bool>(() => GetStaticNonPublicBooleanPropertyValue("System.IO.Strategies.FileStreamHelpers", "UseNet5CompatStrategy"));
```

Comment:
Should we make this throw if the type can't be found?

Reply:
I was thinking about it, but that would require special handling for when the `PlatformDetection` type is used to run Full .NET Framework tests. I decided to just add a test instead. Is that OK?

Reply:
I think Adam meant (please correct me if I'm wrong) that the `PlatformDectection` class, which contains the `_net5CompatFileStream` field, is also consumed by some .NET Framework tests. I can see the file can be built for both net60 and net461:
![image](https://user-images.githubusercontent.com/1175054/129238922-5d2fbc8f-df45-46df-9d2b-89df77a0a27a.png)

What I do not know is what special handling would be required to make the exception work.

Reply:
I've forgotten that it's lazy and only compiled for .NET 4.6.1 (not executed).